website/src/pages/api/username.ts

```diff
@@ -1,19 +1,21 @@
 import { withoutRole } from "src/lib/auth";
 import prisma from "src/lib/prismadb";
+import { getValidDisplayName } from "src/lib/display_name_validation";
 
 /**
  * Updates the user's `name` field in the `User` table.
  */
 const handler = withoutRole("banned", async (req, res, token) => {
   const { username } = req.body;
-  const { name } = await prisma.user.update({
+  let { name } = await prisma.user.update({
     where: {
       id: token.sub,
     },
     data: {
       name: username,
     },
   });
+  name = getValidDisplayName(name, token.sub);
```

Comment:
This is incorrect. You need to check if the username is valid or not and then perform an update.

Reply:
Oh that's true, sorry. Then I should do the same for the other files where that function is used, right? (trollboard, leaderboard...)

Reply:
> Oh that's true, sorry. Then I should do the same for the other files where that function is used, right? (trollboard, leaderboard...)

No need to other files. They look good now.